phpBB/phpbb/db/migration/schema_generator.php

```diff
@@ -84,171 +98,239 @@ public function get_schema()
-						{
-							foreach ($data as $table)
-							{
-								unset($this->tables[$table]);
-							}
-						}
-						else if ($change_type === 'add_columns')
-						{
-							foreach ($data as $table => $add_columns)
-							{
-								foreach ($add_columns as $column => $column_data)
-								{
-									if (isset($column_data['after']))
-									{
-										$columns = $this->tables[$table]['COLUMNS'];
-										$offset = array_search($column_data['after'], array_keys($columns));
-										unset($column_data['after']);
-
-										if ($offset === false)
-										{
-											$this->tables[$table]['COLUMNS'][$column] = array_values($column_data);
-										}
-										else
-										{
-											$this->tables[$table]['COLUMNS'] = array_merge(array_slice($columns, 0, $offset + 1, true), array($column => array_values($column_data)), array_slice($columns, $offset));
-										}
-									}
-									else
-									{
-										$this->tables[$table]['COLUMNS'][$column] = $column_data;
-									}
-								}
-							}
-						}
-						else if ($change_type === 'change_columns')
-						{
-							foreach ($data as $table => $change_columns)
-							{
-								foreach ($change_columns as $column => $column_data)
-								{
-									$this->tables[$table]['COLUMNS'][$column] = $column_data;
-								}
-							}
-						}
-						else if ($change_type === 'drop_columns')
-						{
-							foreach ($data as $table => $drop_columns)
-							{
-								if (is_array($drop_columns))
-								{
-									foreach ($drop_columns as $column)
-									{
-										unset($this->tables[$table]['COLUMNS'][$column]);
-									}
-								}
-								else
-								{
-									unset($this->tables[$table]['COLUMNS'][$drop_columns]);
-								}
-							}
-						}
-						else if ($change_type === 'add_unique_index')
-						{
-							foreach ($data as $table => $add_index)
-							{
-								foreach ($add_index as $key => $index_data)
-								{
-									$this->tables[$table]['KEYS'][$key] = array('UNIQUE', $index_data);
-								}
-							}
-						}
-						else if ($change_type === 'add_index')
-						{
-							foreach ($data as $table => $add_index)
-							{
-								foreach ($add_index as $key => $index_data)
-								{
-									$this->tables[$table]['KEYS'][$key] = array('INDEX', $index_data);
-								}
-							}
-						}
-						else if ($change_type === 'drop_keys')
-						{
-							foreach ($data as $table => $drop_keys)
-							{
-								foreach ($drop_keys as $key)
-								{
-									unset($this->tables[$table]['KEYS'][$key]);
-								}
-							}
-						}
-						else
-						{
-							var_dump($change_type);
-						}
-					}
-					unset($migrations[$migration_key]);
-				}
-				else if ($check_dependencies)
-				{
-					$this->dependencies = array_merge($this->dependencies, $open_dependencies);
-				}
+				throw new \UnexpectedValueException("$migration_class contains undefined schema changes: $change_type.");
```

Comment:
Yeah, all valid change types are defined in the `$column_map` array. And now you will get an exception instead of the `var_dump` if the change type is undefined.